optuna/storages/_journal/storage.py

```diff
@@ -93,8 +98,23 @@ def __init__(self, log_storage: BaseJournalLogStorage) -> None:
         self._replay_result = JournalStorageReplayResult(self._worker_id_prefix)
 
         with self._thread_lock:
+            if isinstance(self._backend, BaseJournalLogSnapshot):
+                self._backend.load_snapshot(self.restore_replay_result)
             self._sync_with_backend()
 
+    def restore_replay_result(self, snapshot: bytes) -> None:
+        try:
+            r: Optional[JournalStorageReplayResult] = pickle.loads(snapshot)
+        except Exception as e:
+            raise SnapshotRestoreError("Failed to restore `JournalStorageReplayResult`.") from e
```

Comment:
How about just ignoring the error and returing `None` instead of raising an error?

This is because,
- IMO, the pickle format is fragile and difficult to keep compatibilities for future Optuna versions.
  - For example, if the module path or field names of `JournalStorageReplayResult` are changed by internal refactoring, the above `pickle.loads()` would fail when `pickle.dumps()` was called with the class before th refactoring (and users could not load the old studies).
  - I'm not a pickle export but I think that, unlike RDBs, it's difficult to provide migrations for pickled data.
- This snapshot data is optional. 
  - Users can restore the complete state without the snapshot (at the expense of loading time).

Reply:
I figured out https://github.com/optuna/optuna/pull/4102/files#diff-66d47f11c6b3c15222471e34c2253dac831d687c73d29f58e13b0a0f9d9a5d95R104-R106 handles the failure case.